pkg/userdata/helper/download_binaries_script.go

```diff
@@ -0,0 +1,56 @@
+package helper
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+)
+
+const (
+	downloadBinariesTpl = `mkdir -p /opt/bin/
+mkdir -p /var/lib/calico
+mkdir -p /etc/kubernetes/manifests
+mkdir -p /etc/cni/net.d
+mkdir -p /opt/cni/bin
+
+# cni
+if [ ! -f /opt/cni/bin/loopback ]; then
+    wget -O /opt/cni-plugins.tar.gz https://github.com/containernetworking/plugins/releases/download/v0.6.0/cni-plugins-amd64-v0.6.0.tgz
+    mkdir -p /opt/cni/bin /etc/cni/net.d
+    tar -xvf /opt/cni-plugins.tar.gz -C /opt/cni/bin/
```

Comment:
Can we please do this and all other archive downloads to come like this:

```
mkdir -p /opt/cni/bin /etc/cni/net.d
curl -L https://github.com/containernetworking/plugins/releases/download/v0.6.0/cni-plugins-amd64-v0.6.0.tgz|tar -xvzC /opt/cni/bin -f -
```

to avoid writing an archive to disk we will never touch again?